src/index.ts

```diff
@@ -1688,13 +1702,17 @@ export class CachedMerklePatriciaTree<K, V> extends MerklePatriciaTree<K, V> {
    *
    * @returns value corresponding to the key if present; null if otherwise
    */
-  getFromCache(key: K, nodeMap: Map<bigint, MerklePatriciaTreeNode<V>>): V {
+  getFromCache(
```

Comment:
Eh, I actually like the version where you could pass in the set:

```typescript
getFromCache( key: K, bagNodesUsed : Set<bigint> = new Set(), ...nodeMap: Array<Map<bigint, MerklePatriciaTreeNode<V>>>) : V
```

That way I could just pass in an existing set of nodes used over and over again, and it doesn't change the return type

Reply:
Okay cool! Slight change: I cannot overload the function with both optional argument and infinite arguments. So, I propose:

getFromCache( key: K, bagNodesUsed : Set<bigint> | undefined, ...nodeMap: Array<Map<bigint, MerklePatriciaTreeNode<V>>>) : V

If you don't want to track the nodes read in a read from the cached-tree; you should set bagNodesUsed as undefined. And if you want to track the nodes read; you can send the same bagNodesUsed over and over again